astropy/nddata/nddata.py

```diff
@@ -192,12 +292,76 @@ def mask(self, value):
 
     @property
     def unit(self):
+        """
+        `~astropy.units.Unit` : Unit for the dataset, if any.
+        """
         return self._unit
 
     @property
     def wcs(self):
+        """
+        any type : A world coordinate system (WCS) for the dataset, if any.
+        """
         return self._wcs
 
     @property
     def meta(self):
+        """
+        `dict`-like : Meta information about the dataset, if any.
+        """
         return self._meta
+
+    @property
+    def uncertainty(self):
+        """
+        any type : Uncertainty in the dataset, if any.
+
+        Should have an attribute ``uncertainty_type`` that defines what kind of
+        uncertainty is stored, such as ``std`` for standard deviation or
+        ``var`` for variance. A metaclass defining such an interface is
+        `~astropy.nddata.NDUncertainty` but isn't mandatory.
+
+        TODO: *Must* or *Should* have this ``uncertainty_type`` ? (mwcraig)
+        I implemented UnknownUncertainty to work around that so what is
+        saved in _uncertainty has an uncertainty_type but it's unknown :-)
+        """
+        if isinstance(self._uncertainty, UnknownUncertainty):
+            return self._uncertainty.array
+        else:
+            return self._uncertainty
+
+    @property
+    def uncertainty_type(self):
```

Comment:
Is the idea here that eventually every `NDData` object would have an uncertainty, defaulting to `UnknownUncertainty`, eventually? Otherwise it seems like leaving the `uncertainty_type` a property of the uncertainty would be easier.